apps/waypoints/waypoints.app.js

```diff
@@ -219,10 +490,85 @@ function showNumpad(text, key_, callback) {
         ]}
       ], lazy:true});
   g.clear();
-  numPad.render();  
+  numPad.render();
   update();
 }
 
+function goTo() {
+  cancel_gps = false;
+  Bangle.setGPSPower(1, "waypoints");
+  gps.gps_start = getTime();
+  gps_start = getTime();
+
+  var la = new Layout (
+    {type:"v", c: [
+      {type:"txt", font:"15%", pad:1, fillx:1, filly:1, label:""},
+      {type:"txt", font:"15%", pad:1, fillx:1, filly:1, label:""},
+      {type:"txt", font:"15%", pad:1, fillx:1, filly:1, label:""},
+      {type:"h", c: [
+        {type:"btn", font:"15%", pad:1, fillx:1, filly:1, label: " Done", cb:l=>{ stopGps(); mainMenu(); }}
+      ]}
+    ], lazy:true});
+  g.clear();
+  la.render();
+
+  updateGoto();
+}
+
+function show(pin) {
+  var i = wp[pin];
+  var l = fmt.fmtPos(i);
+  arrow.name = i.name;
+  arrow.waypoint = i;
+
+  var la = new Layout (
+    {type:"v", c: [
+      {type:"txt", font:"15%", pad:1, fillx:1, filly:1, label:arrow.name },
+      {type:"txt", font:"15%", pad:1, fillx:1, filly:1, label:l },
+      {type:"txt", font:"15%", pad:1, fillx:1, filly:1, label:""},
+      {type:"h", c: [
+        {type:"btn", font:"15%", pad:1, fillx:1, filly:1, label: "Go", cb:l=>{ goTo(); }},
+        {type:"btn", font:"15%", pad:1, fillx:1, filly:1, label: "Ok", cb:l=>{ mainMenu(); }}
+      ]}
+    ], lazy:true});
+  g.clear();
+  la.render();
+}
+
+function showCard() {
+  var menu = {
+    "" : {title : "Select WP"},
+    "< Back" : mainMenu
+  };
+  if (Object.keys(wp).length==0) Object.assign(menu, {"No WPs":""});
```

Comment:
Is `wp` always an array? If so we could:
```suggestion
  if (wp.length==0) Object.assign(menu, {"No WPs":""});
```